backend/router.js

```diff
@@ -2,16 +2,142 @@ const express = require('express')
 const app = express()
 const cors = require('cors')
 require('dotenv').config()
+const groq = require("groq-sdk");
+const multer = require('multer');
+const fs = require('fs');
 
 
 
 
 
 app.use(cors())
 app.use(express.json())
-
+app.use((req, res, next) => {
+    res.setHeader("Cross-Origin-Opener-Policy", "same-origin");
+    res.setHeader("Cross-Origin-Embedder-Policy", "require-corp")
+    next();
+});
 const PORT = process.env.PORT || 3000
-console.log('Port:', PORT)
+
+
+const groqInstance = new groq({ apiKey: process.env.GROQ_API_KEY });
+
+app.post('/api/chat', async (req, res) => {
+    const { message, context, lastQuestionCheck, prevIsFollowUp } = req.body;
+
+    let lastQuestion = lastQuestionCheck == "quit" ? true : false
+    try {
+        const doFollowUp = Math.random() > .65 && !lastQuestion ? true : false;
```

Comment:
I feel that this logic can be consolidated

Reply:
Very true, I have made the code more concise.